nb_anacondacloud/static/main.js

```diff
@@ -81,120 +95,329 @@ define(['jquery', 'base/js/dialog'], function ($, dialog) {
+
+        console.log(arguments)
+
+        switch(jqXHR.status){
+          case 401:
+            title = 'Unauthorized';
+            notif = [
+              $('<p>').text('You are not authorized to complete this action.' +
+                  ' You may need to run this at the command line:'),
+              $('<pre>').text('anaconda login')
+            ];
+            break;
+          case 500:
+            title = label;
+            notif = [
+              $('<p>').text('An error ocurred in the Notebook Server'),
+              $('<p>').text('You may need to re-install nb_anacondacloud')
+            ];
+            break;
+          default:
+            title = 'Error: ' + label;
+            notif = [
+              $('<p>').text('An error ocurred in the Notebook Server'),
+            ]
+        }
+
+        Jupyter.notification_area.get_widget('notebook').
+            danger(title, 4000);
+
+        body = $('<div>').append(notif);
+
         dialog.modal({
             title: title,
             body: body,
             buttons : {
-                "OK": {}
+                OK: {}
             }
         });
-    };
+    }
+
 
-    var publishButton = function() {
-        if (!IPython.toolbar) {
-            $([IPython.events]).on("app_initialized.NotebookApp", publishButton);
+    function updateToolbar() {
+        if (!Jupyter.toolbar) {
+            $([Jupyter.events])
+              .on('app_initialized.NotebookApp', updateToolbar);
             return;
         }
-        if ($("#publish_notebook").length === 0) {
-            IPython.toolbar.add_buttons_group([
-                {
-                    'label'   : 'Publish your notebook into Anaconda.org',
-                    'icon'    : 'fa-cloud-upload',
-                    'callback': configureUpload,
-                    'id'      : 'publish_notebook'
-                }, {
-                    'label'   : 'Visit your notebook',
-                    'icon'    : 'fa-cloud',
-                    'callback': visitNotebook,
-                    'id'      : 'visit_notebook'
-                }
-            ]);
+        if ($('#publish_notebook').length === 0) {
+            Jupyter.toolbar.add_buttons_group([{
+                label: 'Publish your notebook into Anaconda.org',
+                icon: 'fa-cloud-upload',
+                callback: configureUpload,
+                id: 'publish_notebook'
+            }, {
+                label: 'Visit your notebook',
+                icon: 'fa-cloud',
+                callback: visitNotebook,
+                id: 'visit_notebook'
+            }]);
         }
         updateVisitLink();
-    };
+    }
+
+    function load_jupyter_extension() {
+        updateToolbar();
+    }
 
-    var load_ipython_extension = function () {
-        publishButton();
-    };
 
+    // export new and old API entrypoints
     return {
-        load_ipython_extension : load_ipython_extension,
+      load_ipython_extension: load_jupyter_extension,
```

Comment:
The API is still load_ipython_extension:
https://github.com/jupyter/notebook/blob/7abc4e8825c792409e83f6e93492d9ecd217beef/notebook/static/base/js/utils.js

But I thought there was a discussion of deprecating that... could be wrong. will revert.